From 118922c593f86de60d4130d88dc748b4cade4557 Mon Sep 17 00:00:00 2001
From: Eric Meddaugh <eric@meddaughs.com>
Date: Mon, 12 Nov 2018 15:40:42 -0500
Subject: [PATCH 2/2] libbt: Fix logic check when FW_PRE_PATCH is not used.

* If FW_PRE_PATCH is not set prepatch_name will end up NULL.

Change-Id: I27482d37a8871ac8739c47f7869b5a8e4e42347d
---
 src/hardware.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hardware.c b/src/hardware.c
index 8b21b33..7d7e0c5 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -561,7 +561,7 @@ static uint8_t hw_config_findpatch(char *p_chip_id_str)
                 {
                     /* Make sure it's not the prepatch */
                     prepatch_name = strrchr(fw_prepatch_name, '/');
-                    if(strcmp(++prepatch_name, dp->d_name))
+                    if(prepatch_name == NULL || strcmp(++prepatch_name, dp->d_name))
                     {
                         ALOGI("Found patchfile: %s/%s", \
                             fw_patchfile_path, dp->d_name);
-- 
2.17.1

